tokio/src/time/driver/entry.rs

```diff
@@ -543,6 +543,10 @@ impl TimerEntry {
         mut self: Pin<&mut Self>,
         cx: &mut Context<'_>,
     ) -> Poll<Result<(), super::Error>> {
+        if self.driver.is_shutdown() {
```

Comment:
Are we concerned about the additional `inner.upgrade()` inside `is_shutdown` on every timer poll here?

Reply:
The `inner.upgrade()` is only on the io driver. This call is in the time driver which ends here https://github.com/EmbarkStudios/tokio/blob/rt-handle-block-on/tokio/src/time/driver/mod.rs#L452.